### ISSUE TITLE
HDR Color Profile minTML resets to 0.01 nits. Also SDR Gamma question.

### ISSUE THREAD

**Issue by Jason-GitH** [Maassoft/ColorControl#308]
The minTML and minCLL both rest to 0.01 nits. I confirmed this by testing in VESA Display HDR Test app.

I also wanted to confirm something about the SDR gamma. The power setting (default 2.2) only affect the "Pure Power" option in the pull-down correct? Piecewise will override this value to sRGB piecewise gamma, and BT.1886 will override to 2.4, correct?

**Comment by Austinzveare** [comment]
I also noticed the min value, and I also couldn't see any difference between any gamma setting, not even something like 2.6.

**Comment by Maassoft** [comment]
> The minTML and minCLL both rest to 0.01 nits. I confirmed this by testing in VESA Display HDR Test app.

Yes, I've noticed this as well. The values are correctly set in the profile, but I think when retrieving the values through the Windows API (they are not directly read from the active profile) they get changed or it's some kind of Windows issue (like the HDR calibration app not going lower than 0.01 nits initially). Maybe Windows still doesn't handle the absolute zero well.

There's also an issue with rebooting and resuming the system from stand-by: the settings seem to be reverted to other defaults (MinTML/CLL 0.03 and Max to 1200 on my system) and the color profile needs to be reapplied for the values to be correct again. That's probably what the "set_maxtml" tool wants to fix as well. Maybe I'll incorporate these fixes into CC. The changes to the gamma curve are however applied without reactivating the profile again.

> I also wanted to confirm something about the SDR gamma. The power setting (default 2.2) only affect the "Pure Power" option in the pull-down correct? Piecewise will override this value to sRGB piecewise gamma, and BT.1886 will override to 2.4, correct?

Yes, I wanted to disable the fields when not using "Pure Power", but haven't managed to do so.

> I also noticed the min value, and I also couldn't see any difference between any gamma setting, not even something like 2.6.

Changing the gamma is immediately noticeable on my system. Is the checkbox "Use my settings for this device" checked in the Windows Color Management app? ColorControl should display a dialog to enable this. Also validate whether the color profiles are correctly listed under the "ICC Profiles (Advanced Color)" header. What graphics card are you using?

**Comment by Jason-GitH** [comment]
> > The minTML and minCLL both rest to 0.01 nits. I confirmed this by testing in VESA Display HDR Test app.
> 
> Yes, I've noticed this as well. The values are correctly set in the profile, but I think when retrieving the values through the Windows API (they are not directly read from the active profile) they get changed or it's some kind of Windows issue (like the HDR calibration app not going lower than 0.01 nits initially). Maybe Windows still doesn't handle the absolute zero well.
> 
> There's also an issue with rebooting and resuming the system from stand-by: the settings seem to be reverted to other defaults (MinTML/CLL 0.03 and Max to 1200 on my system) and the color profile needs to be reapplied for the values to be correct again. That's probably what the "set_maxtml" tool wants to fix as well. Maybe I'll incorporate these fixes into CC. The changes to the gamma curve are however applied without reactivating the profile again.
> 

I haven't tested this [utility ](https://github.com/ledoge/set_maxtml) myself but it was created because of recent Windows behavior changing minTML to 0.01 nits or ignoring maxTML. People have reported it works successfully at keeping minTML at 0 nits but has to be run after every restart.

**Comment by Austinzveare** [comment]
I figured out why I wasn't seeing any change. Nvidia needs to be set to Accurate color mode, not Reference. Now I see obvious gamma changes. I also use the set_maxtml app to set 0 nits because of the HDR calibration issue.

What would the ideal settings for this be and how would I easily change it back to default Windows HDR gamma, assuming that's what it should be set to for HDR games?
![Screenshot 2023-12-30 100107](https://github.com/Maassoft/ColorControl/assets/124917407/b07b0e50-5d31-445a-b7cb-e009b5b18a47)

**Comment by DitherMan420** [comment]
@Maassoft, can you take another look at the mincll value that your new tool is setting? It shows mincll as 0.01, whereas Dogelition's "Set_maxTML" tool properly sets it to 0.0000 nits. Same with CRU. You should also consider joining the HDR Den discord server. There are a lot of developers there and I think most of that server probably uses your ColorControl tool :-)

**Comment by ValeZAA** [comment]
>It used to in the past as well, but something changed within the past few months (I think with 23h2). 

What changed in Windows 11 23H2? I do not get it.

Also note that Moments 4 is not 23H2. 23H2 has 22631 before the dot.

**Comment by Maassoft** [comment]
> @Maassoft, can you take another look at the mincll value that your new tool is setting? It shows mincll as 0.01, whereas Dogelition's "Set_maxTML" tool properly sets it to 0.0000 nits. Same with CRU. So it's clearly possible to get Windows to correctly show 0.000 nits. It used to in the past as well, but something changed within the past few months (I think with 23h2). So there's definitely a way to get it to work since Doge's tool literally does it :-).
> 
> You should also consider joining the HDR Den discord server. There are a lot of developers there and I think most of that server probably uses your ColorControl tool :-).
> 
> https://github.com/ledoge/set_maxtml
> 
> https://discord.gg/4muSZyznNh

The profiles are created correctly, it's probably just Windows that screws it up somewhere. But I've already created a workaround in my internal test build of CC: after applying a color profile, CC looks at the min/max luminance values of the profile and will then directly call the same API's as the "Set_maxTLL" tool does. In this way the values are reported correctly and you still have control over the values via the properties of the color profile. Then there's still an issue when resuming the system, but I can probably fix that as well.
